### PR TITLE
saving trained models and analysing gridCV scores

### PR DESCRIPTION
**Pull request by elenafillo**
Added the following capabilities:

-  Saving a model during training, and loading it to make predictions
```python
train_baseline_model("MHD", model_type="mlp", ......, save_model=True, save_folder=cfg.models_path, save_suffix="example_run")
```
will save the model to `path/ml-baselines/models/MHD/mlp_model_2026-04-07_13-58_example_run.joblib`, which then can be loaded 
```python
model, extra_info = load_baseline_model("MHD", "mlp", suffix="example_run")
scaler, time_shift_hours = extra_info["scaler"], extra_info["model_inputs"]["time_shift_hours"]
# use the model to predict
y, y_pred, scores  = predict_baselines(site="MHD",  model=model, time_shift_hours=time_shift_hours, scaler=scaler)
```
- Saving the scores of a grid search
```python
train_baseline_model_grid_search(model_type="gradient_boosting", .... return_cv_scores=True, save_cv_scores=True,  save_suffix="test_run")
````
will save the scores as a csv to `path/ml-baselines/models/MHD/cv_results_MHD_gradient_boosting_test_run.csv`, which can then be loaded and inspected
```python
from ml_baselines.modelling.cv_analysis import load_cv_results, get_top_params, plot_param_effects, plot_score_heatmap
df = load_cv_results(site="MHD", save_suffix="test_run", model_type="gradient_boosting")
# print the parameters for the best three models in terms of precision
best_precision_params = get_best_params(df, n=3, metric="precision")
print(best_precision_params)
# plot how the scores change for each parameter (see the plot below)
fig, axes = plot_param_effects(df, metric="all")
plt.show()
```
- Other improvements:
    - `train_baseline_model` now only evaluates on the test set if `evaluate_on_test=True`
    - if balancing or resampling the training dataset with `train_baseline_model`, it will also output the scores on the original training dataset (so that the results are comparable to the validation/test set)
    - `get_train_test_data` now works as expected with test periods that overlap with train/validation periods - the overlapping periods get removed
    - `train_baseline_model_grid_search` now also evaluates different sample_weights, passed through the `data_kwargs` for example `data_kwargs = {"time_shift_hours":[[6], [6,12]], "sample_weights":[None, 2, 5]}`

Example output of plot_param_effects:
image

### CODE REVIEW

File: ml_baselines/modelling/predict.py [hunk 1]
```diff
@@ -9,6 +12,34 @@
...
+        model_path = model_path.with_name(f"{model_type}_model_{timestamp}*.joblib")
+    elif suffix is not None:
+        model_path = model_path.with_name(f"{model_type}_model_*{suffix}.joblib")
+        
+    model_file = glob.glob(str(model_path))
+    if not model_file:
+        raise FileNotFoundError(f"No model found at {model_path}")
+    model_dict = joblib.load(model_file[-1])  # Load the most recent model file
+    print(f"Loaded model from {model_file[-1]}")
+    return model_dict['model'], model_dict['info']
```
[the remark this answers is not in the document]
Comment (elenafillo) [reply ml_baselines/modelling/predict.py:42]: @copilot apply changes based on [this feedback](https://github.com/openghg/ml-baselines/pull/12#discussion_r3058985651)

File: ml_baselines/modelling/cv_analysis.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+    Returns:
+        DataFrame with all columns as-is, plus recalculated rank_test_<metric>
+        columns that are global across all data_kwarg groups.
+    """
+    filename = f"cv_results_{site}_{model_type}_model.csv" if save_suffix is None else f"cv_results_{site}_{model_type}_{save_suffix}.csv"
+    filepath = Path(save_path) / site / filename
+
+    if not filepath.exists():
+        raise FileNotFoundError(f"CV results file not found: {filepath}")
```
[the remark this answers is not in the document]
Comment (elenafillo) [reply ml_baselines/modelling/cv_analysis.py]: @copilot apply changes based on [this feedback](https://github.com/openghg/ml-baselines/pull/12#discussion_r3058985684)

File: ml_baselines/modelling/train.py [hunk 3]
```diff
@@ -335,6 +360,10 @@ def train_baseline_model(site, model_type="mlp",
         return_scores (bool): Whether to return the evaluation scores as a dictionary.
         return_scaler (bool): Whether to return the fitted scaler used for normalising input features.
         verbose (bool): Whether to print verbose output.
+        save_model (bool): Whether to save the trained model.
+        save_folder (str): The folder where the model should be saved. It will be saved in a subfolder named after the site, with a filename based on the model type and current timestamp.
+        save_suffix (str, optional): A suffix to append to the saved model filename. If None, no suffix will be added.
+        random_seed (int): The random seed to use for reproducible results.
```
[the remark this answers is not in the document]
Comment (elenafillo) [reply ml_baselines/modelling/train.py:367]: @copilot apply changes based on [this feedback](https://github.com/openghg/ml-baselines/pull/12#discussion_r3058985603)
